New serverless pattern - DocumentDB Event Source to Lambda

webster5197 opened the pull request:
*Issue #2132 

*Description of changes:*
New SAM Template that creates an AWS Lambda function to process events in an Amazon DocumentDB (with MongoDB compatibility) change stream by configuring an Amazon DocumentDB cluster as an event source 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

webster5197 commented:
Thank you for the suggestions. I have submitted new commit
Modified the grammatical changes suggested
Added Dynamic Secret Manager values
Edited Inline Code to External
Modified IAM role with dedicated resources

webster5197 commented on docdb-lambda/README.md:
````diff
@@ -0,0 +1,159 @@
+    
+    Create the MongoDB repository file 
+    ```
+    echo -e "[mongodb-org-4.0] \nname=MongoDB Repository\nbaseurl=https://repo.mongodb.org/yum/amazon/2013.03/mongodb-org/4.0/x86_64/\ngpgcheck=1 \nenabled=1 \ngpgkey=https://www.mongodb.org/static/pgp/server-4.0.asc" | sudo tee /etc/yum.repos.d/mongodb-org-4.0.repo
````
Tried to explain. This was picked up from the AWS Tutorial docs

bfreiberg commented on docdb-lambda/README.md:
```diff
@@ -0,0 +1,160 @@
+    * Enter the VPC ID where the resources needs to be created
+    * Enter the First Private Subnet ID that belongs to the above VPC
+    * Enter the Second Private Subnet ID that belongs to the above VPC
+    * Enter the Username for AWS Secrets manager to authenicate with Amazon DocumentDB
+    * Enter the Password for AWS Secrets manager to authenicate with Amazon DocumentDB
```
```suggestion
    * Enter the Username for AWS Secrets manager to authenticate with Amazon DocumentDB
    * Enter the Password for AWS Secrets manager to authenticate with Amazon DocumentDB
```

bfreiberg commented on docdb-lambda/README.md:
```diff
@@ -0,0 +1,160 @@
+    * Enter a name for your DocumentDB Instance
+    * Enter the DocumentDB Instance Class (Eg. db.t3.medium)
+    * Enter the VPC ID where the resources needs to be created
+    * Enter the First Private Subnet ID that belongs to the above VPC
+    * Enter the Second Private Subnet ID that belongs to the above VPC
```
```suggestion
    * Enter the first private subnet ID that belongs to the above VPC
    * Enter the second private subnet ID that belongs to the above VPC
```

bfreiberg commented:
Thanks for the update, there are some minor typos but otherwise looks good.

bfreiberg commented:
One more thing, please rename the folder to `docdb-lambda-nodejs-cdk`

webster5197 commented:
Thank you for the suggestions. 
Corrected the requested typo changes.
Also changed the folder named to docdb-lambda-nodejs-sam as I am using SAM and not CDK.

bfreiberg commented:
Thanks for your contribution. A Developer Advocate will merge the changes and publish the pattern to [serverlessland.com](http://serverlessland.com/).
